configuration/configuration_test.go

```diff
@@ -33,15 +33,14 @@ var configStruct = Configuration{
 	},
 	Storage: Storage{
 		"s3": Parameters{
-			"region":        "us-east-1",
-			"bucket":        "my-bucket",
-			"rootdirectory": "/registry",
-			"encrypt":       true,
-			"secure":        false,
-			"accesskey":     "SAMPLEACCESSKEY",
-			"secretkey":     "SUPERSECRET",
-			"host":          nil,
```

Comment:
Silly question; if host and port were no valid options; should this test have failed? Were they valid before (and did we neglect to add a validation step when they were removed?)

Reply:
This test simply test the functions that resolve env REGISTRY_DRIVER_PARAMETERS into config.s3["parameters"]. It doesn't test or check for valid parameters for a specific driver.

This PR only update parameters to the supported ones according to today's implementation.

Validating valid parameters will require some way for the driver to expose valid options to a validator function which I'm unaware of any currently that can be used in this test.

Reply:
Ah, right; I didn't open the file locally.

So, given that these values are just to test the marshalling/unmarshalling and interaction between env-vars and config files, I wonder if it would be better to make that more clear, and to use actual dummy values for these to take away any confusing / wrong assumption that they may matter / are validated.

I checked out your branch locally this time, and made a quick change to do something along those lines. I pushed it as a branch; can open as a PR against your PR-branch if we think that approach makes sense; https://github.com/kaovilai/distribution/compare/s3-no-hostport...thaJeztah:s3_no_hostport_suggestions?expand=1

@milosgajdos @kaovilai WDYT?

Reply:
Merged your suggestions.